packages/api/lambdas/kinesis-consumer.js

```diff
@@ -34,37 +35,48 @@ async function getKinesisRules(event) {
 }
 
 /**
- * `createOneTimeRules` creates new rules with the same data as a kinesis-type rule, except the type is modified to 'onetime'.
+ * `createOneTimeRules` creates new rules with the same data as a kinesis-type rule,
+ * except the type is modified to 'onetime'.
  *
- * @param {array} kinesisRules list of rule objects
- * @returns {array} Array of promises for model.create
+ * @param {Array} kinesisRules - list of rule objects
+ * @param {Object} eventObject - kinesis message input
+ * @returns {Array} Array of promises for model.create
  */
-async function createOneTimeRules(kinesisRules) {
+async function createOneTimeRules(kinesisRules, eventObject) {
```

Comment:
This is not scalable at its current form. We should update this function to only generate the cumulus message and pass it to the SF starter queue for each incoming message.